docs/docs/server/openapi.yaml

```diff
@@ -164,39 +218,109 @@ components:
           description: |
             Set to `true` if the user liked the response, `false` if the user didn't like the response.
             No value if user didn't rate the response.
+
             Note that only messages with `role: "assistant"` can be rated.
+        userComment:
+          type: string
+          description: |
+            Comment from the user clarifying why they liked or disliked
+            the message. No value if user didn't comment on the response.
+
+            Note that only messages with `role: "assistant"` can be commented on.
         createdAt:
           type: integer
           description: |
             Unix timestamp for when the message was sent.
         references:
-          type: array
+          $ref: "#/components/schemas/References"
+    References:
+      type: array
+      description: |
+        List of references to external resources that are relevant
+        to the message. For example, if the message is a response to
+        a user's question about how to connect, the list of
+        references could include links to the documentation, dev
+        center articles, or a video lesson on MongoDB University.
+      items:
+        type: object
+        required: [title, url]
+        properties:
+          title:
+            type: string
+            description: |
+              Title of the reference. This is what shows up in
+              rendered links. For example, "How to connect to
+              MongoDB Atlas".
+          url:
+            type: string
+            format: uri
+            description: |
+              URL to the reference. For example, "https://mongodb.com/docs/atlas/getting-started".
+    MessageResponseStreamEvent:
+      oneOf:
+        - $ref: "#/components/schemas/StreamEventDelta"
+        - $ref: "#/components/schemas/StreamEventReferences"
+        - $ref: "#/components/schemas/StreamEventFinished"
+      description: Item in data stream when streaming a response message.
+    StreamEvent:
```

Comment:
I don't think we need this abstract event - we're overwriting all the fields in every instance we use it